BlueprintUI/Sources/Blueprint View/ElementPreview.swift

```diff
@@ -0,0 +1,240 @@
+/// Uhhh
+/// -----
+/// You're probably asking...
+/// Why the `!arch(i386)` check above? Turns out, a compiler bug!
+/// SwiftUI is only available on 64 bit devices, but the `canImport` check erroneously
+/// finds it when building to target iOS 10 devices. Until we drop iOS 10, this part of the check is also required.
+///
+/// Details
+/// --------
+/// It's important that you keep the `PreviewProvider` in the same file as the element that you are editing.
+///
+/// Why? Xcode uses a new feature called "Dynamic Replacement" to re-compile the source file you are editing,
+/// and inject it back into the running app which drives the preview. This only works on the level of a single
+/// file – if your preview and element live in separate files, Xcode needs to recompile your entire module
+/// which will slow down preview updates greatly.
+///
+/// You can learn more about Xcode previews here: https://nshipster.com/swiftui-previews/
+///
+/// Requirements
+/// --------
+/// You must be running Xcode 11 and Catalina to take advantage of live previews.
+/// They do not work on Mojave. Your selected simulator must also be an iOS 13 device.
+///
+@available(iOS 13.0, *)
+public struct ElementPreview : View {
+    
+    // MARK: Properties
+    
+    /// A provider which returns a new element.
+    public typealias ElementProvider = () -> Element
+    
+    /// The types of previews to include in the Xcode preview.
+    private var previewTypes : [PreviewType]
+    
+    /// The provider which vends a new element.
+    private var provider : ElementProvider
+    
+    // MARK: Initialization
+    
+    
+    /// Creates a new `ElementPreview` with several common devices that your users may use.
+    public static func commonDevices(with provider : @escaping ElementProvider) -> Self {
+        return Self(
+            with: [
+                .device(.iPhoneSE),
+                .device(.iPhone8),
+                .device(.iPhone8Plus),
+                .device(.iPhoneXs),
+                .device(.iPhoneXsMax),
+                
+                .device(.iPadPro_9_7),
+                .device(.iPadPro_10_5),
+                .device(.iPadPro_12_9_3),
+            ],
+            with: provider
+        )
+    }
+    
+
+    /// Creates a new `ElementPreview` with the provided preview type.
+    /// If you do not pass a preview type, `.thatFits` is used.
+    public init(
+        with previewType : PreviewType = .thatFits,
+        with provider : @escaping ElementProvider
+    ) {
+        self.init(
+            with: [previewType],
+            with: provider
+        )
+    }
+    
+    /// Creates a new `ElementPreview` with the provided preview types.
+    ///
+    /// You can pass as many preview types as you would like to see your
+    /// element rendered in those different environments.
+    ///
+    /// If you do not pass a preview type, `.thatFits` is used.
+    public init(
+        with previewTypes : [PreviewType] = [.thatFits],
```

Comment:
I don’t think it really matters, but any reason for the default arg when that just duplicates the same init as the single preview type? Wondering if that confuses auto complete at all. Probably fine though!

Reply:
Fair! Will remove; I’d added this method first without rethinks once I added the singular one above.

Also realizing I should assert in here that at least one value is passed, or maybe log since crashes in previews are hard to debug